common/src/main/org/jetbrains/lincheck/trace/Utils.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2025 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.lincheck.trace
+
+import org.jetbrains.lincheck.util.isJavaLambdaClass
+
+/**
+ * Replaces nested class dollars (if present) from string with dots.
+ */
+fun replaceNestedClassDollar(nestedClassRepresentation: String): String {
+    val before = nestedClassRepresentation.substringBefore('$')
+    val after = nestedClassRepresentation.substringAfter('$', "")
```

Comment:
BTW, these `$` signs are an artifact of our not-fully-correct internal-to-canonical class name conversion (implemented in `toCanonicalClassName()` method):

https://www.baeldung.com/java-class-name#3-inner-classes


Let's create an issue for that, because it is not the first time we encounter this.